src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Unix.cs

```diff
@@ -60,12 +60,11 @@ public sealed partial class TarWriter : IDisposable
 
             entry._header._mode = (status.Mode & 4095); // First 12 bits
 
-            entry.Uid = (int)status.Uid;
-            entry.Gid = (int)status.Gid;
+            entry._header._uid = (int)status.Uid;
+            entry._header._gid = (int)status.Gid;
 
-            // TODO: Add these p/invokes https://github.com/dotnet/runtime/issues/68230
-            entry._header._uName = "";// Interop.Sys.GetUName();
-            entry._header._gName = "";// Interop.Sys.GetGName();
+            entry._header._uName = Interop.Sys.GetUName(status.Uid) ?? string.Empty;
+            entry._header._gName = Interop.Sys.GetGName(status.Gid) ?? string.Empty;
```

Comment:
You probably want to cache these in a Dictionary to avoid looking them up each time over?

And, I think the user may want to be able to control the the user id and name that get stored in the tar file.

I think it would be meaningful if the API had option classes that control the creation behavior, and extraction behavior. These classes can be extended later on, rather than adding new overloads to the existing methods.

Reply:
> You probably want to cache these in a Dictionary to avoid looking them up each time over?

Good point. If I already know the `uname` and `gname` associated to a `uid` and `gid`, there's no need to retrieve them again.

> And, I think the user may want to be able to control the the user id and name that get stored in the tar file.

This code path is only reached if the user calls `TarWriter.WriteEntry(string fileName, string? entryName)`. If they wish to choose the `uname`, `gname`, `uid`, `gid` to other values than the default, they can create the `TarEntry` instance manually, fill out its fields, and use the other overload `TarWriter.WriteEntry(TarEntry entry)`.

> I think it would be meaningful if the API had option classes that control the creation behavior, and extraction behavior. These classes can be extended later on, rather than adding new overloads to the existing methods.

I would welcome that 🙂. We can discuss it in a separate issue.